### PR TITLE
perf(remote-config): merge blob data via byte concatenation

### PR DESCRIPTION
**Pull request by ajpallares**
### Motivation
`mergeItemsBlobData` decoded every blob into an intermediate value and re-serialized the keyed dictionary before decoding it into the target type. Since each blob is already valid JSON, that round-trip is unnecessary.

### Description
Assemble the merged object by concatenating the raw blob bytes verbatim (`{"<itemKey>":<blobBytes>,...}`) and decode once. No behavior change; existing `RemoteConfigManagerTests` cover the merge paths.

### Advantages
- Less CPU: drops N blob parses plus the full re-serialization.
- Lower peak memory: no intermediate object graph.
- Preserves blob bytes exactly, avoiding number round-tripping that can lose precision.

---

> [!NOTE]
> **Medium Risk**
> Merge behavior now depends on blobs being valid JSON fragments and on manual envelope assembly; edge cases are covered by new tests but this path feeds decoded remote config payloads.
> 
> **Overview**
> **`mergeItemsBlobData`** no longer decodes each blob through **`AnyDecodable`** and re-serializes a dictionary; it now builds the merged JSON object by stitching **`{`**, **`JSONEncoder`**-escaped item keys, and each blob’s raw bytes, then decodes once into **`T`**.
> 
> This keeps blob payloads verbatim (including numeric precision and leading/trailing whitespace) and cuts CPU/memory from the old decode→encode→decode path. New unit tests cover JSON-escaped and Unicode keys, whitespace-wrapped blobs, and three-way merges.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ccae0c37adb417697e012a1962814f5f252ad41d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: Tests/UnitTests/Networking/RemoteConfig/RemoteConfigManagerTests.swift [hunk 1]
```diff
@@ -884,6 +884,80 @@ final class RemoteConfigManagerTests: TestCase {
...
+
+        let value = try await self.manager.mergeItemsBlobData(
+            for: .workflows,
+            itemKeys: ["wf1"],
+            as: SingleMergedWorkflowPayload.self
+        )
+
+        expect(value) == SingleMergedWorkflowPayload(wf1: .init(value: "spaced"))
+    }
+
```
Comment (rickvdl) [inline Tests/UnitTests/Networking/RemoteConfig/RemoteConfigManagerTests.swift:1046]: Maybe we could add one test here with multiple keys and blobs? It's probably already tested through other codepaths, but just making sure it is explicitly tested here

**Review by rickvdl — APPROVED** [review]
This is a really interesting approach, I like it. Thanks for doing this! 🙌 

One tiny suggestion for a test we could add, but LGTM!

File: Tests/UnitTests/Networking/RemoteConfig/RemoteConfigManagerTests.swift (hunk 1 above)
Comment (ajpallares) [reply Tests/UnitTests/Networking/RemoteConfig/RemoteConfigManagerTests.swift:1046]: Great point! Added in https://github.com/RevenueCat/purchases-ios/pull/7163/commits/659e8cd4dcf8cd5c02543aba3bb1cbde57ed3e5c

**Comment by ajpallares** [comment]
@RCGitBot please test
